src/overlays/actors/ovl_En_Bubble/z_en_bubble.c

```diff
@@ -32,73 +31,399 @@ const ActorInit En_Bubble_InitVars = {
+        COLSHAPE_JNTSPH,
+    },
+    ARRAY_COUNT(sJntSphElementsInit),
+    sJntSphElementsInit,
 };
 
-// sColChkInfoInit
-static CollisionCheckInfoInit2 D_808A0758 = { 1, 2, 25, 25, MASS_IMMOVABLE };
-
-#endif
-
-extern ColliderJntSphElementInit D_808A0700[2];
-extern ColliderJntSphInit D_808A0748;
-extern CollisionCheckInfoInit2 D_808A0758;
-
-extern UNK_TYPE D_06001000;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bubble/func_8089F4E0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bubble/func_8089F59C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bubble/func_8089F5D0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bubble/func_8089F5F4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bubble/func_8089F660.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bubble/func_8089F8BC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bubble/func_8089F908.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bubble/func_8089F95C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bubble/func_8089F9E4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bubble/func_8089FA54.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bubble/func_8089FF30.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bubble/func_8089FFCC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bubble/func_808A005C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bubble/EnBubble_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bubble/EnBubble_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bubble/func_808A029C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bubble/func_808A0350.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bubble/func_808A03A0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bubble/func_808A03E8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bubble/EnBubble_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bubble/EnBubble_Draw.s")
+static CollisionCheckInfoInit2 sColChkInfoInit = { 1, 2, 25, 25, MASS_IMMOVABLE };
+
+static Vec3f sEffectAccel = { 0.0f, -0.5f, 0.0f };
+
+static Color_RGBA8 sEffectPrimColor = { 255, 255, 255, 255 };
+
+static Color_RGBA8 sEffectEnvColor = { 150, 150, 150, 0 };
+
+void EnBubble_SetDimensions(EnBubble* this, f32 dim) {
+    f32 a;
+    f32 b;
+    f32 c;
+    f32 d;
+
+    this->actor.flags |= ACTOR_FLAG_1;
+    Actor_SetScale(&this->actor, 1.0f);
+    this->actor.shape.yOffset = 16.0f;
+    this->graphicRotSpeed = 16.0f;
+    this->graphicEccentricity = 0.08f;
+    this->expansionWidth = dim;
+    this->expansionHeight = dim;
+    a = Rand_ZeroOne();
+    b = Rand_ZeroOne();
+    c = Rand_ZeroOne();
+    this->unk210 = 1.0f;
+    this->unk214 = 1.0f;
+    d = (a * a) + (b * b) + (c * c);
+    this->unk1F8.x = a / d;
+    this->unk1F8.y = b / d;
+    this->unk1F8.z = c / d;
```

Comment:
Set but never used? Was this unk vec used in OOT?

Reply:
Same for OoT, never used. The same goes for unk210 and unk214